### ISSUE TITLE
Different behaviour between cypari2 and parigp : factorial

### ISSUE THREAD

**Issue by vinklein** [defeo/cypari2#42]
parigp version return a real and cypari2 an integer

parigp
```
break[3]> type(factorial(10))
"t_REAL"
```
cypari2
```
pari.type(pari.factorial(10))
Out[]: "t_INT"
```
I don't know if it's a design choice or a real issue.

**Comment by jdemeyer** [comment]
> I don't know if it's a design choice or a real issue.

Probably both :-)

In the GP language, one has the `!` operator for integer factorial:
```
gp> 10!
%1 = 3628800
```
This syntax simply doesn't exist in Python, so it was decided to use the `factorial()` method for this. But, this way we do override the `factorial()` function from GP. I think we should have two methods: `factorial()` for the real (in both senses of the word) factorial function and something else (maybe `fact()` or `factorial_int()`) for the integer factorial.

Note that you can get access to the GP `factorial()` function this way:
```
sage: from cypari2.pari_instance import Pari_auto
sage: Pari_auto.factorial(pari, 10)
3628800.00000000
```

**Comment by vinklein** [comment]
Ok thanks.

**Comment by videlec** [comment]
fixed in #45
